libr/include/r_types.h

```diff
@@ -13,6 +13,20 @@
 #undef __UNIX__
 #undef __WINDOWS__
 
+#define R_IN /* do not use, implicit */
+#define R_OWN /*own*/
+#define R_OUT /*out*/
+#define R_INOUT /*inout*/
+#define R_NULL /*null*/
```

Comment:
shall we mark this as default behaviour? otherwise it's unclear whether a dev should expect null or not when nothing is used.

Reply:
maybe R_NULLABLE is a better name :? not sure which should be the default but i would expect nothing can be null unless specified

Reply:
I like `R_NULLABLE` and I'm fine with assuming everything should be non-null, unless specified.